account_general_ledger_report/models/general_ledger_report.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+import datetime
+from openerp import models, api, fields, _
+
+
+class AccountGeneralLedgerReport(models.Model):
+    _name = 'account.general.ledger.report'
+
+    name = fields.Char(
+        string='Name',
+        readonly=True,
+    )
+    fiscalyear_id = fields.Many2one(
+        'account.fiscalyear',
+        string='Fiscal Year',
+        readonly=True,
+    )
+    target_move = fields.Selection(
+        [('posted', 'All Posted Entries'),
+         ('all', 'All Entries')],
+        string='Target Moves',
+        readonly=True,
+    )
+    reconcile_cond = fields.Selection(
+        [('all', 'All Items'),
+         ('open_item', 'Open Items'),
+         ('reconciled', 'Full Reconciled')],
+        string=" ",
+        readonly=True,
+    )
+    account_ids = fields.Many2many(
+        'account.account',
+        string="Accounts",
+        required=True,
+    )
+    amount_currency = fields.Boolean(
+        string='With Currency',
+        default=True,
+    )
+    line_ids = fields.One2many(
+        'account.general.ledger.line',
+        'report_id',
+        string='Details'
+    )
+
+    @api.model
+    def _get_moves(self, fiscalyear_id, target_move, reconcile_cond,
+                   account_ids, amount_currency):
+        Period = self.env['account.period']
+        Move = self.env['account.move.line']
+        Fiscalyear = self.env['account.fiscalyear'].browse(fiscalyear_id)
+        periods = []
+        # All moves, begin of this year until date_stop
+        domain = [('period_id.fiscalyear_id', '=', fiscalyear_id),
+                  ('account_id', 'in', account_ids.ids)]
+        if target_move == 'posted':
+            domain.append(('move_id.state', '=', 'posted'))
+        if reconcile_cond == 'open_item':
+            domain.append(('reconcile_id', '=', False))
+        if reconcile_cond == 'reconciled':
+            domain.append(('reconcile_id', '!=', False))
+        moves = Move.search(domain)
+
+        self._cr.execute("""
+            SELECT periods.period_id, periods.name
+            FROM (SELECT ap.id AS period_id, ap.name
+                FROM account_period ap
+                LEFT JOIN account_fiscalyear af
+                ON ap.fiscalyear_id = af.id
+                WHERE ap.special = 'f' AND af.code = '%s') periods
+                LEFT JOIN
+                (SELECT ap.id AS period_id, ap.name
+                FROM account_move_line aml
+                LEFT JOIN account_period ap ON aml.period_id = ap.id
+                LEFT JOIN account_fiscalyear af ON ap.fiscalyear_id = af.id
+                WHERE ap.special = 'f' AND af.code = '%s'
+                GROUP BY ap.id) rpt
+                ON periods.period_id = rpt.period_id
+                ORDER BY periods.name
+        """ % (Fiscalyear.code, Fiscalyear.code))
```

Comment:
When join or where clause, please use "id".
Joining and Where with .id is faster than using character field.

Reply:
In this case, I think it is better to use SQL Injection, rather than String Injectoin.
i.e.,
> self._cr.execture("""select * from abc where my_id = %s and my_string = %s""", (my_id, my_string))
Not
> self._cr.execture("""select * from abc where my_id = %s and my_string = '%s' """ % (my_id, my_string))

For query, do not use String Injection if it is not necessary.